api/handlers/service_binding_test.go

```diff
@@ -72,476 +72,555 @@ var _ = Describe("ServiceBinding", func() {
 	Describe("POST /v3/service_credential_bindings", func() {
 		var payload payloads.ServiceBindingCreate
 
-		BeforeEach(func() {
-			requestMethod = http.MethodPost
-			requestPath = "/v3/service_credential_bindings"
-			requestBody = "the-json-body"
-
-			payload = payloads.ServiceBindingCreate{
-				Relationships: &payloads.ServiceBindingRelationships{
-					App: &payloads.Relationship{
-						Data: &payloads.RelationshipData{
-							GUID: "app-guid",
-						},
-					},
-					ServiceInstance: &payloads.Relationship{
-						Data: &payloads.RelationshipData{
-							GUID: "service-instance-guid",
+		When("creating a service binding of type key", func() {
```

Comment:
This newly introduced `When` is a copy of the existing create binding tests. Intead of putting existing tests under `When("creating a service binding of type app"`, I would recommend introduce a context that verifies that unprocessable entity error is returned when creating a binding of type key to a user provided service:


```golang
Describe("POST /v3/service_credential_bindings", func() {
  When("binding to a user provided service instance", func() {
    When("the binding is of type key", func(){
      It("returns an unprocessable entity error", func(){...})
    })
```